status/condition_set.go

```diff
@@ -237,3 +240,62 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// unstructuredAdapter is an embedded struct type that embeds the upstream type. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+// FromUnstructured makes the assumption that the status conditions are found on status.conditions
+// path. If they are not found then we return nil.
+func FromUnstructured(u *unstructured.Unstructured) Object {
+	_, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+	if err != nil || !found {
+		return nil
+	}
+	return &unstructuredAdapter{Unstructured: u}
+}
+
+func (u *unstructuredAdapter) GetConditions() []Condition {
+	var conditions []Condition
+	c, _, _ := unstructured.NestedSlice(u.Object, "status", "conditions")
+	for _, condition := range c {
+		var newCondition Condition
+		cond := condition.(map[string]interface{})
+		jsonStr, err := json.Marshal(cond)
+		if err != nil {
+			continue
+		}
+		if err = json.Unmarshal(jsonStr, &newCondition); err != nil {
+			continue
+		}
+		conditions = append(conditions, newCondition)
+	}
+	return conditions
+}
+
+func (u *unstructuredAdapter) SetConditions(conditions []Condition) {
+	uc, _, _ := unstructured.NestedSlice(u.Object, "status", "conditions")
+	lo.Map(conditions, func(condition Condition, _ int) []interface{} {
+		uc = append(uc, map[string]interface{}{
+			"type":               condition.Type,
+			"status":             string(condition.Status),
+			"message":            condition.Message,
+			"reason":             condition.Reason,
+			"observedGeneration": strconv.Itoa(int(condition.ObservedGeneration)),
+			"lastTransitionTime": condition.LastTransitionTime.Time.String(),
+		})
+		return uc
+	})
+	unstructured.SetNestedSlice(u.Object, uc, "status", "conditions")
+}
+
+func (u *unstructuredAdapter) StatusConditions() ConditionSet {
+	var conditionType []string
+	lo.Map(u.GetConditions(), func(condition Condition, _ int) []string {
```

Comment:
```suggestion
	conditionTypes := lo.Map(u.GetConditions(), func(condition Condition, _ int) []string {
```

lo.Map outputs a new array so that you don't have to pre-declare and append within the function itself. You just return what you want the elements to look like in the array and `lo` constructs it all for you